src/store.js

```diff
@@ -1,14 +1,90 @@
+import axios from 'axios';
+import Config from './config.js';
 
 var store = {
     config  : {},
-    user    : {},
-    trader  : {},
+    user    : {"id" : 1},
+    trader  : {"id" : 1},
     vouchers: [],
+    serverVouchers: []
 };
 
-store.addVoucherCode = function(voucherCode)
-{
-    return this.vouchers.push(voucherCode);
+store.getServerVouchers = function() {
+    if (!navigator.onLine) {
+        return false;
+    }
+    this.apiGet('traders/1/vouchers', function(response) {
+        var newVouchers = response.data.map(function(v) {
+            return v.code;
+        });
+        store.mergeServerVouchers(newVouchers);
+    });
+    return true;
+};
+
+store.apiGet = function(route, callback) {
+    if (!route.match(/^\//)) {
+        route = '/' + route;
+    }
+    axios.get(Config.apiBase + route)
+        .then(callback)
+        .catch(this.logAJAXErrors);
+};
+
+store.mergeServerVouchers = function(newVouchers) {
+    // em-biggen newVouchers
+    Array.prototype.push.apply(newVouchers, this.serverVouchers);
```

Comment:
as with javascript, it's *a way* to merge them. acts a bit like array_merge. concat apparently can do odd things.